common/djangoapps/student/views.py

```diff
@@ -1612,6 +1612,8 @@ def _do_create_account(form, custom_form=None):
         user=user,
         **{key: form.cleaned_data.get(key) for key in profile_fields}
     )
+    # this field is added identify the user's domain through which he registered.
+    profile.created_on_site = configuration_helpers.get_value('site_domain', '')
```

Comment:
We should pass the current site to this function and use `site.domain` to set this value instead of looking it up in site configuration. You can access the current site from the calling view using `request.site`.